### PR TITLE
Testing Ruby with RSpec: Split apart Ruby's Connect Four assignment into 3 separate files

### LINKED ISSUE #29632: Ruby: Splitting the Connect4 Project into two separate assignments
### Checks

- [x] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [x] Would you like to work on this issue?

### Describe your suggestion

I believe that the Connect4 project's page is really overwhelming and cluttered. It basically has two assignments baked into one. I don't really see any benefits or reasoning behind this. It isn't even the final assignment of the section, as there's still a final project to write after this.

Splitting it apart would make progress tracking a bit more friendly, and it would lessen the burden of tackling what is essentially a lesson, assignment, and then another assignment in a single repo, into something more manageable/less daunting. 

### Path

Ruby / Rails

### Lesson Url

https://www.theodinproject.com/lessons/ruby-connect-four

### (Optional) Discord Name

tokisuno

### (Optional) Additional Comments

i love top

### PR DESCRIPTION
**Pull request by tokisuno**
## Because
I found that the Connect Four section of the Ruby path was very overwhelming. I proposed to split it up to make it less overwhelming.

## This PR
- Creates new sections
  - practice_rspec.md
  - revisiting_tictactoe.md
  - project_connect_four.md

## Issue

Closes #29632

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/testing_ruby_with_rspec/practice_rspec.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### Introduction
+
+Before getting into any big projects, you should have some practice with writing tests in RSpec. Learning is best done through doing after all! There are a lot of tests to go through and you have to refer to multiple files, but make sure to not feel too overwhelmed. We're all in this together!
+
+For this section, you're first going to complete a lot of exercises involving test-writing, and then finally implementing tests into an old you've written.
```
Comment (JoshDevHub) [inline ruby/testing_ruby_with_rspec/practice_rspec.md:5]:
```suggestion
For this section, you're first going to complete a lot of exercises involving test-writing, and then finally implementing tests into an old project you've written.
```
Missing word here.

File: ruby/testing_ruby_with_rspec/project_connect_four.md [hunk 2]
```diff
@@ -1,21 +1,6 @@
...
-  1. Write tests for your [Tic Tac Toe project](/lessons/ruby-tic-tac-toe). In this situation, it's not quite as straightforward as just coming up with inputs and making sure the method returns the correct thing. You'll need to make sure the tests that determine victory or loss conditions are correctly assessed.
-      1. Start by writing tests to make sure players win when they should, e.g. when the board reads X X X across the top row, your `#game_over` method (or its equivalent) should trigger.
-      1. Test each of your critical methods to make sure they function properly and handle edge cases.
-      1. Use mocks/doubles to isolate methods to make sure that they're sending back the right outputs.
-
-</div>
+At this point, you've written tests for older projects of yours, and have completed a plethora of other test-writign assignments to familiarize yourself with RSpec. You're now ready to create a new project with TDD in mind!
```
Comment (JoshDevHub) [inline ruby/testing_ruby_with_rspec/project_connect_four.md:3]:
```suggestion
At this point, you've written tests for older projects of yours, and have completed a plethora of other test-writing assignments to familiarize yourself with RSpec. You're now ready to create a new project with TDD in mind!
```

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
This looks good. Just a couple small nits/typos to fix and then it'll be good to approve.

**Review by JoshDevHub — CHANGES_REQUESTED** [review]
There are also a few markdown lint issues you'll need to fix. If you have trouble figuring any of them out, just let me know.

One of these is related to the difference in requirements between "projects" and "lessons." The linter assumes that your `practice_rspec.md` and `revisiting_tictactoe.md` files are lesson files because they don't have a `project_` prefix. This means they expect a lesson structure to the markdown.

You can resolve this error by adding prefixing those file names with `project_` just like the connect four file is.

**Comment by JoshDevHub** [comment]
And on last comment: in the original issue, I mentioned there will have to be some steps taken to actually get the content on the site. Did you want to take that work on? No big deal if not, and we can get someone else to do it.

**Comment by JoshDevHub** [comment]
@tokisuno Any update on this?

**Comment by tokisuno** [comment]
> @tokisuno Any update on this?

Sorry man. Went through a breakup and relapsed hard on gaming. Almost 90 hours in the past 2 weeks logged onto CS2. Putting this onto my tasklist and will get to it this week.

**Comment by tokisuno** [comment]
Someone else can finish this, as I have no interest in finishing this. I moved on to working through FullStackOpen for the time being.
